src/schemas.jl

```diff
@@ -440,19 +474,18 @@ function _generate_schema_version_definitions(schema_version::SchemaVersion, par
 end
 
 function _generate_validation_definitions(schema_version::SchemaVersion)
-    field_violation_check_statements = Expr[]
-    for (fname, ftype) in pairs(required_fields(schema_version))
-        fname = Base.Meta.quot(fname)
-        push!(field_violation_check_statements, quote
-            S = $Legolas.accepted_field_type(sv, $ftype)
-            result = $Legolas._check_for_expected_field(ts, $fname, S)
-            isnothing(result) || return $fname => result
-        end)
-    end
+    SV = Base.Meta.quot(typeof(schema_version))
     return quote
-        function $(Legolas).find_violation(ts::$(Tables).Schema, sv::$(Base.Meta.quot(typeof(schema_version))))
-            $(field_violation_check_statements...)
```

Comment:
i'd actually keep `field_violation_check_statements` and the original inlined generated definition for `find_violation`, which is nicely non-allocating / optimizable

then, to approach defining `find_violations`, i'd construct something like `field_violation_push_statements` (or w/e, probably a better name exists lol) in the same loop that constructs `field_violation_check_statements`; very DRYable since the only difference is `push!(violations, x)` instead of `return x` (where `x` is `$fname => result`) 

then generate the `find_violations` definition similar to the singular version

Reply:
done